test/run_tests_model_gen_and_load.py

```diff
@@ -68,15 +68,23 @@ def create_test_dir(
         shutil.copy(str(file_to_copy), str(test_dest_file))
 
 
-def generate_model(
+def generate_model_and_weights(
     test_id: int,
     command: str,
     working_dir: Path,
     color_enum: Type[Union[Color, NoColor]] = Color,
 ) -> None:
     print(f"{color_enum.LIGHT_CYAN}id: {test_id}, command: {command}{color_enum.ENDC}")
     vw = vowpalwabbit.Workspace(command, quiet=True)
-
+    weights_dir = working_dir / "../test_weights"
```

Comment:
why is test_weights above working_dir?

Reply:
working_dir here the model directory, so the test_weights directory lives next to it

Reply:
I think it would be better to make the parent directory the working_dir. That way you don't need to go out of your directory scope for any of the work here.

Another option would be to move the model_dir up a level, and make the entire structure look like
```
root_dir
  + outputs
    + model_dir
    + test_weights
```

Then outputs would be the working_dir in this case

Reply:
updated to this format